tests/core/test_open_in_process.py

```diff
@@ -14,6 +14,32 @@
 )
 
 
+@pytest.mark.asyncio
+async def test_SIGINT_on_method_using_run_in_executor():
+    async def wait_forever():
+        import threading
+        stopped = threading.Event()
+        loop = asyncio.get_event_loop()
+        try:
+            await loop.run_in_executor(None, stopped.wait)
```

Comment:
I believe that this might be why it hangs forever.  IIUC:

- `wait_forever` will run in a separate process
- In that process, `stopped.wait` will be run in the default executor which is the `ThreadPoolExecutor`

Since threads are not cancellable and `stopped.wait` will never complete, that suggests that whatever thread it is running in will also not complete.  `asyncio` may correctly exit the loop, but I have recollections of processes hanging because there are threads still running that haven't exited....   So I would be looking at a way to verify that the thread is still running.  Maybe instead of using a `threading.Event()` you could instead use a synchronous function that does something observable  periodically in a loop but never exits.  Maybe repeatedly touching a file, or writing something to STDOUT....  Then you could see whether the thread is still running by checking if the thread is still  doing it's thing.

Reply:
That's the issue indeed! I feel like we need to somehow document it to prevent headaches in the future, but at the same time I'm not sure it belongs in `open_in_process()`'s docstring, so maybe we can keep the test with the explanation I've added to it?